service/service.go

```diff
@@ -233,7 +233,13 @@ var installStartRetryAttempts = utils.AttemptStrategy{
 // InstallAndStart installs the provided service and tries starting it.
 // The first few Start failures are ignored.
 func InstallAndStart(svc ServiceActions) error {
-	logger.Infof("Installing and starting service %+v", svc)
+	service, ok := svc.(Service)
+	if !ok {
+		return errors.Errorf("specified service has no name %+v", svc)
```

Comment:
Ideally we'd update the interface so we could avoid this, but since all the implementations implement Name it should be fine.